consensus/core/src/leader_scoring.rs

```diff
@@ -138,173 +69,203 @@ impl ReputationScores {
     }
 }
 
-/// UnscoredSubdag represents a collection of subdags across multiple commits.
-/// These subdags are considered unscored for the purposes of leader schedule
-/// change. On a leader schedule change, reputation scores will be calculated
-/// based on the dags collected in this struct. Similar graph traversal methods
-/// that are provided in DagState are also added here to help calculate the
-/// scores.
-pub(crate) struct UnscoredSubdag {
+/// ScoringSubdag represents the scoring votes in a collection of subdags across
+/// multiple commits.
+/// These subdags are "scoring" for the purposes of leader schedule change. As
+/// new subdags are added, the DAG is traversed and votes for leaders are recorded
+/// and scored along with stake. On a leader schedule change, finalize reputation
+/// scores will be calculated based on the votes & stake collected in this struct.
+pub(crate) struct ScoringSubdag {
     pub(crate) context: Arc<Context>,
-    pub(crate) commit_range: CommitRange,
-    pub(crate) committed_leaders: Vec<BlockRef>,
-    // When the blocks are collected form the list of provided subdags we ensure
-    // that the CommittedSubDag instances are contiguous in commit index order.
-    // Therefore we can guarantee the blocks of UnscoredSubdag are also sorted
-    // via the commit index.
-    pub(crate) blocks: BTreeMap<BlockRef, VerifiedBlock>,
+    pub(crate) commit_range: Option<CommitRange>,
+    // Only includes committed leaders for now.
+    // TODO(arun): Include skipped leaders as well
```

Comment:
Decided on moving this to the next PR as we will have to modify the committer output which is a bit sensitive and could use its own review. Also need to consider that we cannot score skipped votes with the vote scoring strategy as that would cause problems during upgrade. So we would need to have some logic in here to not score skipped leaders which can make this messy. Or we can do a separate upgrade to include skipped leaders after certified vote goes live.